packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts

```diff
@@ -804,6 +804,11 @@ export class CoinbaseWalletProvider extends EventEmitter implements Web3Provider
         return this._wallet_watchAsset(params);
     }
 
+    this.diagnostic?.log(EVENTS.METHOD_NOT_IMPLEMENTED, {
```

Comment:
This will emit a `METHOD_NOT_IMPLEMENTED` event for all json rpc calls (i.e. if the dapp is using the web3 provider to call `eth_getBalance` and so on).  I don't think that's correct since those methods are implemented - by the json rpc endpoint.

I think what we want is to catch the response from `makeEthereumJSONRPCRequest` and if it's an unimplemented method error, then emit the event. 

Something like
```
    return relay.makeEthereumJSONRPCRequest(request, this.jsonRpcUrl).catch((err) => {
      if (err.code === standardErrorCodes.rpc.methodNotFound) {
        this.diagnostic?.log(EVENTS.METHOD_NOT_IMPLEMENTED, {
          method: request.method,
          sessionIdHash: this._relay ? Session.hash(this._relay.session.id) : undefined,
        });
      }

      throw err;
    });
```

Reply:
Or to be extra safe
```
    return relay.makeEthereumJSONRPCRequest(request, this.jsonRpcUrl).catch((err) => {
      if (err.code === standardErrorCodes.rpc.methodNotFound || err.code === standardErrorCodes.rpc.methodNotSupported) {
        this.diagnostic?.log(EVENTS.METHOD_NOT_IMPLEMENTED, {
          method: request.method,
          sessionIdHash: this._relay ? Session.hash(this._relay.session.id) : undefined,
        });
      }

      throw err;
    });
```

Reply:
Good catch. I should have noticed this.

Reply:
@hieronymus777 thanks for pointing this out. i totally missed.
will update soon. cc @cb-jake

Reply:
https://github.com/coinbase/coinbase-wallet-sdk/pull/1040